### ISSUE TITLE
Self links are incorrectly encoding URL scheme

### ISSUE THREAD

**Issue by kgutwin** [marshmallow-code/marshmallow-jsonapi#171]
Recently-merged PR #147 introduced quoting for URLs such as self_url. My codebase includes fully-qualified URLs such as `https://example.com`. Unfortunately, these are now being incorrectly quoted to `https%3A//example.com`. 

```
>>> from marshmallow_jsonapi import Schema, fields
>>> class TestSchema(Schema):
...     id = fields.Str()
...     class Meta:
...         type_ = 'post'
...         self_url = 'https://example.com/posts'
... 
>>> class TestObject:
...     def __init__(self):
...         self.id = 4
... 
>>> result = TestSchema().dump(TestObject())
>>> result.data['links']['self']
'https%3A//example.com/posts'
```

Would it be possible to apply the URL encoding only to the path and query portions?

**Comment by darwinyip** [comment]
This seems like a pretty critical bug.

**Comment by scottwernervt** [comment]
The pull request should be undone (no quoting). If a user needs quoting, they should override the url generate / build methods by subclassing `schema.Schema` and `fields.Relationship`.

**Comment by grantHarris** [comment]
Here you go https://github.com/marshmallow-code/marshmallow-jsonapi/pull/184

**Comment by franklinchou** [comment]
Any movement on this?

**Comment by grantHarris** [comment]
I'll be looking at it again tonight

**Comment by franklinchou** [comment]
Any movement on this?

**Comment by sloria** [comment]
The url quoting has been reverted. Released in 0.21.0.
